tests/flows/erc20_token_staking.go

```diff
@@ -0,0 +1,183 @@
+package flows
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	warpMessages "github.com/ava-labs/avalanchego/vms/platformvm/warp/messages"
+	warpPayload "github.com/ava-labs/avalanchego/vms/platformvm/warp/payload"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/teleporter/tests/interfaces"
+	"github.com/ava-labs/teleporter/tests/utils"
+	. "github.com/onsi/gomega"
+)
+
+// Registers a erc20 token staking validator on a subnet. The steps are as follows:
+// - Deploy the ERCTokenStakingManager
+// - Initiate validator registration
+// - Deliver the Warp message to the P-Chain (not implemented)
+// - Aggregate P-Chain signatures on the response Warp message
+// - Deliver the Warp message to the subnet
+// - Verify that the validator is registered in the staking contract
+// Delists the validator from the subnet. The steps are as follows:
+// - Initiate validator delisting
+// - Deliver the Warp message to the P-Chain (not implemented)
+// - Aggregate P-Chain signatures on the response Warp message
+// - Deliver the Warp message to the subnet
+// - Verify that the validator is delisted from the staking contract
+func ERC20TokenStakingManager(network interfaces.LocalNetwork) {
+	// Get the subnets info
+	cChainInfo := network.GetPrimaryNetworkInfo()
+	subnetAInfo, _ := utils.GetTwoSubnets(network)
+	_, fundedKey := network.GetFundedAccountInfo()
+	pChainInfo := utils.GetPChainInfo(cChainInfo)
+
+	signatureAggregator := utils.NewSignatureAggregator(
+		cChainInfo.NodeURIs[0],
+		[]ids.ID{
+			subnetAInfo.SubnetID,
+			ids.Empty, // Primary network subnet ID
+		},
+	)
+
+	// Deploy the staking manager contract
+	stakingManagerAddress, stakingManager, _, erc20 := utils.DeployAndInitializeERC20TokenStakingManager(
+		context.Background(),
+		fundedKey,
+		subnetAInfo,
+		pChainInfo,
+	)
+
+	//
+	// Register a validator
+	//
+	var validationID ids.ID // To be used in the delisting step
+	stakeAmount := uint64(1e18)
+	weight := uint64(1e6) // stakeAmount / 1e12
```

Comment:
can set the weight with that math instead of hardcoded, incase stakeAmoutn changes